app/models/ability.rb

```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # can :read, [Post, Comment], author: user
+    return unless user.present?
+
+    can :destroy, [Post, Comment]
+    return unless user.role?
+
+    can :destroy, [Post, Comment]
```

Comment:
- [ ] You didn't yet implement your authorization as expected. for now, even the user who is not the owner of a post had the authorization to delete a comment. Kindly, fix this issue. You can take a look at this code below to know how to implement it.
```ruby
 class Ability
 include CanCan::Ability

     def initialize(user)

           user ||= User.new # guest user (not logged in)
  
           if user.admin? 
           can :manage, :all
         else
           can :manage, Post, author: user
           can :manage, Comment, author: user
           can :read, :all
         end
     end
 end
```
Kindly, try to fix that issue.